File: C++ Code/007_Problem121.cpp
# include <iostream>
# include <vector>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
    
    int buy = prices[0], profit = 0;

    for(int i=1;i<prices.size();i++){
        if (prices[i]<buy){
            buy = prices[i];
        }else if(profit<prices[i]-buy){
            profit = prices[i]-buy;
        }
    }
    return profit;
    }
};

File: Python Code/007_Problem121.py
# 121. Best Time to Buy and Sell Stock
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock/?envType=study-plan-v2&envId=top-interview-150

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProf = 0
        bCost = prices[0]

        for cDay in range(len(prices)):
            if(prices[cDay] < bCost):
                bCost = prices[cDay]
            if(prices[cDay] - bCost > maxProf ): 
                maxProf = prices[cDay] - bCost
        
        return maxProf
